onnxruntime/core/graph/function.cc

```diff
@@ -315,6 +340,20 @@ FunctionImpl::FunctionImpl(const onnxruntime::Graph& graph,
   if (!cached_op_schema) {
     // Infer a op_schema for stand-alone functions.
     IOTypeConstraintHelper(onnx_func_proto_, op_schema_, input_name_idx_map, output_name_idx_map);
+
+    // add model local functions to "this" graph. model local functions allow nested functions
```

Comment:
I haven't understood the overall flow here. It is unclear why the creation of this FunctionImpl is updating the outer graph. Wouldn't it have been better for some (transitive) caller to have done the right thing in terms of adding these FunctionProtos to the main graph?

Reply:
nit: "allow nested functions" => "allow nested function calls". "nested uninitialized functions" => "nested uninitialized function calls".